sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JDBCOptions.scala

```diff
@@ -111,6 +111,9 @@ class JDBCOptions(
   // the number of partitions
   val numPartitions = parameters.get(JDBC_NUM_PARTITIONS).map(_.toInt)
 
+  // the default number of partitions
+  val defaultNumPartitions = parameters.getOrElse(DEFAULT_NUM_PARTITIONS, "10").toInt
```

Comment:
What is the difference between `JDBC_NUM_PARTITIONS`?

Reply:
Same question here :)
Do we really need a default? If the users want to have multiple partitions, shouldn't they specify this explicitly?

Reply:
> What is the difference between `JDBC_NUM_PARTITIONS`?

@wangyum 
1. partitionColumn, lowerBound, upperBound and numPartitions must be specified together. If an unreasonable numPartitions was specified by users, such as 1, the parallelism is still very small.
2. therefore, we(jdbc) should rezoning partition nums using another config, not JDBC_NUM_PARTITIONS(it is specified by users and its value maybe very small)

Reply:
> Same question here :) Do we really need a default? If the users want to have multiple partitions, shouldn't they specify this explicitly?

@huaxingao 
1. if users want to change partition nums, they only need to specify `DEFAULT_NUM_PARTITIONS` explicitly and do not need to specify any more parameters, jdbc will automatically calculate the upper and lower bounds of partitions using primary key
2. If users do not specify any parameters, then we(jdbc) need a default value to determine the base number of partitions

Reply:
I replied below. I think if this is about default number of partitions, we can just set the default to JDBC_NUM_PARTITIONS flag.